pkg/controlplane/reconcilers/lease.go

```diff
@@ -117,13 +123,23 @@ func (s *storageLeases) RemoveLease(ip string) error {
 	return s.storage.Delete(apirequest.NewDefaultContext(), s.baseKey+"/"+ip, &corev1.Endpoints{}, nil, rest.ValidateAllObjectFunc, nil)
 }
 
+func (s *storageLeases) Destroy() {
+	s.destroyFn()
+}
+
 // NewLeases creates a new etcd-based Leases implementation.
-func NewLeases(storage storage.Interface, baseKey string, leaseTime time.Duration) Leases {
+func NewLeases(config *storagebackend.ConfigForResource, baseKey string, leaseTime time.Duration) (Leases, error) {
+	leaseStorage, destroyFn, err := storagefactory.Create(*config, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating storage factory: %v", err)
+	}
+	var once sync.Once
```

Comment:
why not making it part of the object?

Reply:
it is a naive question, what will be the difference, runtime allocation?

Reply:
I don't have any strong preference - either works for me. And I don't think there is huge difference between those two.

Reply:
it was just curiosity